### DISCUSSION TITLE
Tracking sales in real time

### DISCUSSION THREAD

**Discussion by DarthDavid404** [etsy/open-api#771] [Q&A]
I am trying to accomplish this v2 method in the v3 Api. Does anyone know if this can still be done without Oauth2?

creation_tsz | public | none | float | The date and time the transaction was created, in epoch seconds.
https://www.etsy.com/developers/documentation/reference/transaction

https://developer.etsy.com/documentation/reference#tag/Shop-Receipt-Transactions - I found the method but it says api key and Oauth2 in the authorizations. I am a little confused. Did this method change from public to private?

Also, if I want to get the sales of a specific shop over a specific period (1 day, 7 days, 30 days), do I have to track that shop and store that data in a Database or does the API have the ability to be queried by dates? I can't seem to find that anywhere. 

Finally, is there still a way to check the exact variation and price of the variation sold in v3 without Oauth2?

**Comment by etsyachristensen** [comment]
I'm not entirely clear on what exactly you are trying to do. In V3, you have to use an OAuth2 token as the bearer token and provide your api key in the x-api-key header. That endpoint requires the `transactions_r` scope to be granted. If your app was not granted that permission previously, your users will have to go through the auth consent process first.

For the second question, you would need to query the shop receipts with the max_created and min_created inputs. (https://developer.etsy.com/documentation/reference/#operation/getShopReceipts). Then you can query the transactions by receipt. (https://developer.etsy.com/documentation/reference/#operation/getShopReceiptTransactionsByReceipt).

Without OAuth2 you will never get the receipt and transaction data (where the variation selected will be provided). To be clear, API V2 used OAuth1. API V3 uses OAuth 2.

**Reply by DarthDavid404** [reply]
To be clear, I am building an app to help sellers optimize their listings and find new product ideas through sales data and keyword / tag research.

I am no longer sure if I can do this with the V3 api.

My current objective is to get the top 100 sellers by sales and by revenue over a 24 hour period, a 7 day period, a 30 day period, and all time. By the sound of it, I won't be able to get the revenue data for the top 100 shops on Etsy. Is this accurate? I also don't see a way to retrieve the top 100 Etsy shops by sales.

I also want to implement tag research where someone can type in a tag and get the top 100 most sold listings over a 24 hour period, 7 day period, and 30 day period that use that tag as well as the amount of searches related to that tag over a 12 month period. I can't seem to find search data or listing views anywhere in the api.

I am also trying to use the transaction data to be able to tell when a transaction was created so I can give a daily rundown of the exact times sales occur for a specific keyword so people can utilize social media marketing more effectively to drive sales to their shops. Does the following method show transaction creation or does it show when a listing was renewed, updated, etc... (https://developers.etsy.com/documentation/reference#operation/findAllListingsActive) - creation_timestamp

Also, on findAllListingsActive, I can't seem to find the meaning of "score" anywhere in the documentation.

**Reply by DarthDavid404** [reply]
I am trying to create tools similar to Marmalead.com and Erank.com.

**Reply by Gareth064** [reply]
> To be clear, I am building an app to help sellers optimize their listings and find new product ideas through sales data and keyword / tag research.
> 
> I am no longer sure if I can do this with the V3 api.
> 
> My current objective is to get the top 100 sellers by sales and by revenue over a 24 hour period, a 7 day period, a 30 day period, and all time. By the sound of it, I won't be able to get the revenue data for the top 100 shops on Etsy. Is this accurate? I also don't see a way to retrieve the top 100 Etsy shops by sales.
> 
> I also want to implement tag research where someone can type in a tag and get the top 100 most sold listings over a 24 hour period, 7 day period, and 30 day period that use that tag as well as the amount of searches related to that tag over a 12 month period. I can't seem to find search data or listing views anywhere in the api.
> 
> I am also trying to use the transaction data to be able to tell when a transaction was created so I can give a daily rundown of the exact times sales occur for a specific keyword so people can utilize social media marketing more effectively to drive sales to their shops. Does the following method show transaction creation or does it show when a listing was renewed, updated, etc... (https://developers.etsy.com/documentation/reference#operation/findAllListingsActive) - creation_timestamp
> 
> Also, on findAllListingsActive, I can't seem to find the meaning of "score" anywhere in the documentation.
> 

Why do you think you should be able to get sales transaction data from other people's companies? There's no way I'd want your app knowing the number of sales I've had in a day, how much they were for and on what listings. 

Transaction data also includes the private information of the buyer as well so you'd end up seeing all that data.

Is there any other marketplace that would allow you to get this information publicly from its sellers? If so, let me know, so I can avoid using them 😂

👇 Not a dig against you here
Reading the type of stuff people are asking the API to do on here is crazy. Half the time you'd think the developer is trying to build another market place themselves but using Etsy as their backend.

**Reply by kbasali** [reply]
I agree with @Gareth064 about sales transactions.
But you can calculate how many sales they have made in a day or month.

**Comment by nateswart** [comment]
Correct, as @etsyachristensen mentions above, in v3 you are required to gain permission/authorization from users in order to obtain this information (via OAuth 2).
